posydon/binary_evol/MESA/step_mesa.py

```diff
@@ -1251,29 +1259,39 @@ def __call__(self, binary):
         state_1 = self.binary.star_1.state
         state_2 = self.binary.star_2.state
         event = self.binary.event
-        mass_ratio = self.binary.star_2.mass/self.binary.star_1.mass
+        m1 = self.binary.star_1.mass
+        m2 = self.binary.star_2.mass
+        mass_ratio = m2/m1
         p = self.binary.orbital_period
-        p_max_HMS_grid = 6500
-        if (state_1 == 'H-rich_Core_H_burning'
-                and state_2 == 'H-rich_Core_H_burning' and event == 'ZAMS'
-                and mass_ratio <= 1. and p <= p_max_HMS_grid):
+        if (state_1 == 'H-rich_Core_H_burning' and 
+            state_2 == 'H-rich_Core_H_burning' and 
+            event == 'ZAMS' and
+            self.m1_min <= m1 <= self.m1_max and 
+            self.q_min <= mass_ratio <= self.q_max and 
+            self.p_min <= p <= self.p_max):
             self.flip_stars_before_step = False
             super().__call__(self.binary)
-        elif (state_1 == 'H-rich_Core_H_burning'
-              and state_2 == 'H-rich_Core_H_burning'
-              and event == 'ZAMS' and mass_ratio > 1. and p <= p_max_HMS_grid):
+        elif (state_1 == 'H-rich_Core_H_burning' and 
+              state_2 == 'H-rich_Core_H_burning' and 
+              event == 'ZAMS' and
+              self.m1_min <= m2 <= self.m1_max and
+              1./self.q_min >= mass_ratio > self.q_max and 
```

Comment:
Here we need a `1./self.q_max`, too. Just because we use a defaulted `q_max=1` doesn't mean it will always be that way. Alternatively, we could say `self.q_min <= 1./mass_ratio <= self.q_max`, which I'd find even more intuitive.